pkg/webhook/webhook.go

```diff
@@ -0,0 +1,104 @@
+package webhook
+
+import (
+	"github.com/kubeflow/kfserving/pkg/apis/serving"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha1"
+	"github.com/kubeflow/kfserving/pkg/webhook/admission/kfservice"
+	"k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	webooktypes "sigs.k8s.io/controller-runtime/pkg/webhook/types"
+)
+
+var (
+	// KFServingWebhookServerName is the name for the webhook server for KFServing resources
+	KFServingWebhookServerName = "kfserving-webhook-server"
+
+	// KFServingWebhookServerNamespace is the namespace for the webhook server for KFServing resources
+	KFServingWebhookServerNamespace = "kubeflow-system"
+)
+var log = logf.Log.WithName(KFServingWebhookServerName)
+
+// AddToManager adds all Controllers to the Manager
+// +kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations;validatingwebhookconfigurations,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch;create;update;patch;delete
+func AddToManager(manager manager.Manager) error {
+	server, err := webhook.NewServer("webhook-server", manager, webhook.ServerOptions{
+		Port:    9876,
+		CertDir: "/tmp/cert",
+		BootstrapOptions: &webhook.BootstrapOptions{
+			Secret: &types.NamespacedName{
+				Namespace: KFServingWebhookServerNamespace,
+				Name:      KFServingWebhookServerName + "-secret",
+			},
+			Service: &webhook.Service{
+				Namespace: KFServingWebhookServerNamespace,
```

Comment:
Service  namespace shouldn't be specified by hardcode KFServingWebhookServerNamespace, it should be specified dynamic to keep consistent with the kfservice controller namespace, saying get it from /var/run/secrets/kubernetes.io/serviceaccount/namespace or env var, etc

Reply:
I think we need a constants file for things like this. We probably don't want to pull this from configmap or env though.

Reply:
I think there is use case where kfserving can be deployed as  individual component so we should not restrict to deploy in `kubeflow-system` namespace, also `kustomize` allows user to change namespace, the logic here may break.  I think we can use the `POD_NAMESPACE` env which is already specified on controller StatefulSet manifest https://github.com/kubeflow/kfserving/blob/master/config/manager/manager.yaml#L51.

Reply:
Sounds good. Though I will need to default for `make run`